build-conventions/src/main/java/org/elasticsearch/gradle/internal/conventions/VersionPropertiesLoader.java

```diff
@@ -43,19 +43,17 @@ protected static void loadBuildSrcVersion(Properties loadedProps, ProviderFactor
             );
         }
         String qualifier = providers.systemProperty("build.version_qualifier")
-                .orElse("")
                 .forUseAtConfigurationTime()
-                .get();
+                .getOrElse("");
```

Comment:
For my own education, why is this way better?

Reply:
the forUseAtConfigurationTime() unfortunately isn't applied to the Provider returned I the `orElse` block, which makes the build fail when running with `configuration-cache` enabled. That behaviour is quite unexpected, at least for my understanding. I'm discussing the shortages of this api with the Gradle team

Reply:
having said that the gradle team mentioned that `forUseAtConfigurationTime` will go away in the future

Reply:
> having said that the gradle team mentioned that forUseAtConfigurationTime will go away in the future

Is there going to be a replacement? Does that mean that you can't depend on things like system properties or environment variables at configuration time?